shub/image/build.py

```diff
@@ -49,19 +53,71 @@ def build_cmd(target, version, skip_tests):
             "Dockerfile is not found, please use shub image 'init' command")
-    is_built = False
-    for data in client.build(path=project_dir, tag=image_name, decode=True):
-        if 'stream' in data:
-            utils.debug_log("{}".format(data['stream'][:-1]))
-            is_built = re.search(
-                r'Successfully built ([0-9a-f]+)', data['stream'])
-        elif 'error' in data:
-            click.echo("Error {}:\n{}".format(
-                data['error'], data['errorDetail']))
-    if not is_built:
-        raise shub_exceptions.RemoteErrorException(
-            "Build image operation failed")
+    if utils.is_verbose():
+        build_progress_cls = _LoggedBuildProgress
+    else:
+        build_progress_cls = _BuildProgress
+    events = client.build(path=project_dir, tag=image_name, decode=True)
+    build_progress = build_progress_cls(events)
+    build_progress.show()
     click.echo("The image {} build is completed.".format(image_name))
     # Test the image content after building it
     if not skip_tests:
         test_cmd(target, version)
+
+
+class _LoggedBuildProgress(utils.BaseProgress):
+    """Visualize build progress in verbose mode.
+
+    Output all the events received from the docker daemon.
+    """
+    def handle_event(self, event):
+        if 'stream' in event:
+            self.handle_stream_event(event)
+        if 'error' in event:
+            tqdm.write("Error {}: {}".format(event['error'],
+                                             event['errorDetail']))
+            raise shub_exceptions.RemoteErrorException(
+                "Build image operation failed")
+
+    def handle_stream_event(self, event):
+        utils.debug_log("{}".format(event['stream'][:-1]))
+
+
+class _BuildProgress(_LoggedBuildProgress):
+    """Visualize build progress in non-verbose mode.
+
+    Show total progress bar.
+    """
+
+    def __init__(self, events):
+        super(_BuildProgress, self).__init__(events)
+        self.bar = None
+        self.is_built = False
+
+    def show(self):
+        super(_BuildProgress, self).show()
+        if self.bar:
+            self.bar.close()
+        if not self.is_built:
+            raise shub_exceptions.RemoteErrorException(
+                "Build image operation failed")
+
+    def handle_stream_event(self, event):
+        if BUILT_REGEX.search(event['stream']):
+            self.is_built = True
+            return
+        step_row = BUILD_STEP_REGEX.match(event['stream'])
+        if not step_row:
+            return
+        step_id, total = [int(val) for val in step_row.groups()]
+        if not self.bar:
+            self.bar = utils.create_progress_bar(
+                total=total,
+                desc='layers',
+                # don't need rate here, let's simplify the bar
+                bar_format='{l_bar}{bar}| {n_fmt}/{total_fmt}',
+            )
+        # ignore onbuild sub-steps
+        if step_id > self.bar.pos and self.bar.total == total:
```

Comment:
`self.bar.pos` is a bar position relative to the first bar in the multiple bars output. you wan't to use `self.bar.n`